warehouse/accounts/views.py

```diff
@@ -778,3 +782,54 @@ def edit_profile_button(user, request):
 )
 def profile_public_email(user, request):
     return {"user": user}
+
+
+@view_config(
+    require_methods=["POST"],
+    require_csrf=False,
+    renderer="json",
+    route_name="accounts.github-disclose-token",
+    header="GITHUB-PUBLIC-KEY-IDENTIFIER",
+    # TODO: How to check multiple headers in the predicates ?
+    # header="GITHUB-PUBLIC-KEY-SIGNATURE"
```

Comment:
It is not possible at the moment; I submitted https://github.com/Pylons/pyramid/issues/3575

A custom predicate is needed here to be able to write `headers=("header1", "header2")`
see here: https://raw.githubusercontent.com/Pylons/pyramid/master/docs/narr/urldispatch.rst
(cloudflare issue prevents seeing the docs and github rendering misses the code blocks, so it’s source markup)

Reply:
So you think the best course of action would be for your PR to land on master and then in warehouse (no idea if we're talking days, weeks, years) or build a custom predicate now ?

Reply:
It could be weeks before Warehouse can update to a Pyramid release that includes my PR.
Go for a custom predicate now!

Reply:
Is there a predicate that combines predicates ? In that case maybe I could simply have something like `And(Header(...), Header(...))`

Reply:
Ok, seems I get an easy solution with:
```
class HeadersPredicate:
    def __init__(self, val: List[str], config):
        self.sub_predicates = [
            predicates.HeaderPredicate(subval, config) for subval in val
        ]

    def text(self):
        return ", ".join(sub.text() for sub in self.sub_predicates)

    phash = text

    def __call__(self, context, request):
        return all(sub(context, request) for sub in self.sub_predicates)
```

Reply:
ok done, with tests and all.